sportsref/ncaaf/players.py

```diff
@@ -174,3 +246,67 @@ def awards(self):
         for yr, award in zip(*[iter(results)]*2):
             ret[int(yr)].append(award)
         return dict(ret)
+
+    @sportsref.decorators.memoized
+    def all_annual_stats(self):
+        """Gets yearly all annual stats for the player by grabbing
+        each individual dataset and then merging them for full years.
+        :kind: One of 'R', 'P', or 'B'. Case-insensitive; defaults to 'R'.
+        :returns: Pandas DataFrame with all annual stats.
+        """
+        dfPassing = self.passing()
+        # dfPassing = dfPassing.ix[dfPassing["has_class_full_table"]]
+        dfRushRec = self.rushing_and_receiving()
+        # dfRushRec = dfRushRec.ix[dfRushRec["has_class_full_table"]]
+        dfDefense = self.defense()
+        # dfDefense = dfDefense.ix[dfDefense["has_class_full_table"]]
+        dfReturns = self.punt_kick_returns()
+        # dfReturns = dfReturns.ix[dfReturns["has_class_full_table"]]
+        dfKicking = self.kicking()
+        # dfKicking = dfKicking.ix[dfKicking["has_class_full_table"]]
+        dfScoring = self.scoring()
+        # dfScoring = dfScoring.ix[dfScoring["has_class_full_table"]]
+        # the mergeList declares the common fields to merge on
+        mergeList = ['year', 'school_name', 'conf_abbr', 'class', 'pos', 'g']
+        dfAll = pd.DataFrame(columns=mergeList)
+        if not dfPassing.empty:
+            dfAll = dfAll.merge(dfPassing, 'outer', mergeList)
+        if not dfRushRec.empty:
+            dfAll = dfAll.merge(dfRushRec, 'outer', mergeList)
+        if not dfDefense.empty:
+            dfAll = dfAll.merge(dfDefense, 'outer', mergeList)
+        if not dfReturns.empty:
+            dfAll = dfAll.merge(dfReturns, 'outer', mergeList)
+        if not dfKicking.empty:
+            dfAll = dfAll.merge(dfKicking, 'outer', mergeList)
+        if not dfScoring.empty:
+            dfAll = dfAll.merge(dfScoring, 'outer', mergeList)
+        return dfAll
+
+def get_college_leaders_one_year(year):
```

Comment:
Oh yea, that sounds great. I didn't know where to put this as it is appropriate in a seasons class but didn't have a home right now since that class doesn't exist. I agree with you completely. At first I was going to use this method in my forecasting model, but now I'm not, so I don't mind if it's tossed out for now.